pkg/resourcecollector/resourcecollector.go

```diff
@@ -346,8 +351,15 @@ func (r *ResourceCollector) GetResources(
 					if !ok {
 						return nil, fmt.Errorf("error casting object: %v", o)
 					}
-
-					collect, err := r.objectToBeCollected(includeObjects, labelSelectors, resourceMap, runtimeObject, crbs, ns, allDrivers)
+					var collect bool
+					var err error
+					// If a namespace is present in both namespace list and IncludeResource Object,
+					// IncludeResource takes priority and only those resources are backed up.
+					// If a ns is only present in namespace list, all resources in that ns
+					// is backed up.
+					// With this now a user can choose to backup all resources in a ns and some
+					// selected resources from different ns
+					collect, err = r.objectToBeCollected(objectToInclude, labelSelectors, resourceMap, runtimeObject, crbs, ns, allDrivers)
```

Comment:
nit: Cann't we not skip calling objectToBeCollected directly here, if the objcetToInclude is empty?

Reply:
If objectToInclude is empty, that could also mean that it was not part of IncludeResources but should be considered as a part of a type provided in the list of ResourceTypes.

Reply:
My understanding is that, we will not call GetResources() to collect resource for ResourcType.  We will be calling GetResourcesForType for resourceType case.
But any how, we need to have the other four check that we do in the objectToBeCollected. We might need to call objectToBeCollected, even if the objectToInclude is empty.
```
metadata, err := meta.Accessor(object)	
    if err != nil {		return false, err	}
	if SkipResource(metadata.GetAnnotations())
          {		return false, err	}
	// Skip if we've already processed this object	
	if _, ok := resourceMap[metadata.GetUID()]; ok {		return false, nil	}
	objectType, err := meta.TypeAccessor(object)	if err != nil {		return false, err	}
	```

Reply:
As discussed there is no change needed here as these checks are already done inside `objectToBeCollected()` API.